src/com/sahokia/kos/bustickets/service/BusTicketsService.java

```diff
@@ -0,0 +1,91 @@
+        List<BusTicket> busTickets = new ArrayList<>();
+
+        try (BufferedReader reader = new BufferedReader(new FileReader(filePath))) {
+            String line;
+            while ((line = reader.readLine()) != null) {
+                if (line.trim().isEmpty()) {
+                    continue;
+                }
+                String[] fields = line.split(",");
+                if (fields.length == 4) {
+                    String ticketClassStr = fields[0].trim();
+                    String ticketTypeStr = fields[1].trim();
+                    String startDateStr = fields[2].trim();
+                    String priceStr = fields[3].trim();
+
+                    TicketClass ticketClass = null;
+                    TicketType ticketType = null;
+                    BigDecimal price = null;
+                    LocalDate startDate = null;
+
+                    if (!ticketClassStr.equals("null") && !ticketClassStr.isEmpty()) {
+                        try {
+                            ticketClass = TicketClass.valueOf(ticketClassStr);
+                        } catch (IllegalArgumentException e) {
+                            System.out.println("Invalid ticket class: " + ticketClassStr);
+                        }
+                    }
+
+                    if (!ticketTypeStr.equals("null") && !ticketTypeStr.isEmpty()) {
+                        try {
+                            ticketType = TicketType.valueOf(ticketTypeStr);
+                        } catch (IllegalArgumentException e) {
+                            System.out.println("Invalid ticket type: " + ticketTypeStr);
+                        }
+                    }
+
+                    if (!priceStr.equals("null") && !priceStr.isEmpty()) {
+                        try {
+                            price = new BigDecimal(priceStr);
+                        } catch (NumberFormatException e) {
+                            System.out.println("Invalid price format: " + priceStr);
+                        }
+                    }
+
+                    if (!startDateStr.equals("null") && !startDateStr.isEmpty()) {
+                        try {
+                            startDate = LocalDate.parse(startDateStr, dateFormatter);
+                        } catch (Exception e) {
+                            System.out.println("Invalid start date format: " + startDateStr);
+                        }
+                    }
+                    BusTicket ticket = new BusTicket(ticketClass, ticketType, startDate, price);
+                    busTickets.add(ticket);
+                }
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        busTicketsStorage.addAll(busTickets);
+    }
+
+    public List<BusTicket> getBusTicketsStorage() {
+        return busTicketsStorage;
+    }
+
+    public void validateBusTickets() {
+        this.busTicketsValidator.validateBusTickets(this.busTicketsStorage);
```

Comment:
'this' is not really needed here :)

Reply:
I have removed busTicketStorage from the class, but got the point that we better should avoid unnecessary "this" prefixes.